server/shared/src/main/scala/org/http4s/server/middleware/authentication/NonceF.scala

```diff
@@ -32,16 +32,22 @@ private[authentication] class NonceF[F[_]](
 )
 
 private[authentication] object NonceF {
-  val random = new SecureRandom()
+  private def getRandomData[F[_]: Functor](random: Random[F], bits: Int): F[String] = {
+    val bytes = (bits + 7) / 8
+    random.nextBytes(bytes).map { arr =>
+      if (arr.nonEmpty) {
+        val extraBits = 8 * bytes - bits
+        arr(0) = (arr(0) & ((1 << (8 - extraBits)) - 1)).toByte
+      }
+      new BigInteger(arr).toString(16)
+    }
+  }
```

Comment:
I think this is about the same as `Random.nextBigInt` from our CE2 shim.  We can't add it to `cats.effect.std.Random`, but we could create a derived method for it.

Reply:
To hell with your BigInt.